src/utilities.jl

```diff
@@ -281,6 +281,7 @@ random effects.
     [`update!`](@ref) requires.
 """
 function create_re(sigmas...; corrmat=nothing)
+    @assert all(>(0),sigmas) "all σs must be positive"
```

Comment:
1. `@assert` can be optimized out in certain cases, so there's no guarantee that this will execute
2. On the mixed model side, the restriction is that sigma is greater than or equal to zero (if you're simulating singular fits).
3. I have very mixed feelings about this level of hand holding. While this is certainly an assumption of the code (and should probably be mentioned in the docstring), a user that knowlingly passes a negative variance for simulation can't be trusted with simulation at all. :wink: If it's about catching sigmas that were programmatically generated elsewhere and there was a bug in that elsewhere, then elsewhere should be responsible for its own bug.